### DISCUSSION TITLE
Limit threshold audit command

### DISCUSSION THREAD

**Discussion by jeff-slang** [xavidop/dialogflow-cx-cli#253] [Ideas]
It would be pretty cool to have a command that audits an agent to assert if the agent is near any limits. Here are the docs for [Dialogflow limits](https://cloud.google.com/dialogflow/quotas).

For example, if an agent has 45 flows (limit is 50 flows), then running something like: `cxcli audit-agent-limits --threshold ".75" --agent-name "Agent" --other-params` would have an error because `45/50 > .8`.

**Comment by xavidop** [comment]
that is a great idea, I am wondering if I can get the limits from some Google APIs

**Reply by jeff-slang** [reply]
That's a great point. I took a look and couldn't find anything. Not even a hosted flat file. Not sure how valuable this automation audit would be if the limits get out of sync with any hard-coded values 😞

**Reply by xavidop** [reply]
yeah... Also, people can request quota increments as well so hard coding those limits is not a good practice.

I am going to close this discussion.
